asn1tools/codecs/per.py

```diff
@@ -37,6 +38,31 @@ def is_unbound(minimum, maximum):
             or (maximum > 65535))
 
 
+def to_int(chars):
+    if isinstance(chars, int):
+        return chars
+
+    num = 0
+    byte_array = bytearray(chars)
+    while len(byte_array) > 0:
+        num <<= 8
+        byte = byte_array.pop(0)
+        num += byte
+
+    return num
+
+
+def to_byte_array(num, number_of_bits):
+    byte_array = bytearray()
+
+    while number_of_bits > 0:
+        byte_array.insert(0, (num & 0xff))
+        num >>= 8
+        number_of_bits -= 8
+
+    return byte_array
+
+
```

Comment:
I'm a bit worries about the performance. Encoding and decoding big BMPString (and other string in the same class) will be very slow. Not sure what to do about it.

Reply:
I have done some benchmarking. The current code is approximately 50% slower than the previous code on very large fixed-width strings. Decoding of ~30MB of such strings takes about 15-20 seconds on my machine.
It is possible to trade-off performance for complexity, if this is a concern, and speed up the operation for ascii-encoded strings.

Reply:
Ok, that was not as bad as I expected. I think that's acceptable for now. I'm sure the performance can be improved later without compromising code complexity too much.